wasp/apps/snake.py

```diff
@@ -152,65 +149,95 @@ def update(self):
-            print("Inside 1")
             return True
         for i in range(len(self.body)-1):
             part = self.body[i]
             if (part[0] == x and part[1] == y):
-                print("Inside 2")
                 return True
         return False
 
     def show(self):
         draw = wasp.watch.drawable
-        if self.len == 1: #vanish old and show new
-            draw.fill(x=(self.body[0][0]-self.xdir),y=(self.body[0][1]-self.ydir),w=15,h=15,bg=0x0000)
-            draw.fill(x=self.body[0][0]+1,y=self.body[0][1]+1,w=13,h=13,bg=0xffff)
-        else: # vanish last and show first
-           draw.fill(x=self.body[0][0],y=self.body[0][1],w=15,h=15,bg=0x0000)
-           draw.fill(x=self.body[-1][0]+1,y=self.body[-1][1]+1,w=13,h=13,bg=0xffff)
-        #for i in range(self.len):
-        #   draw.fill(x=self.body[i][0]+1,y=self.body[i][1]+1,w=13,h=13,bg=0xffff)
-
+        draw.fill(x=self.oldtail[0],y=self.oldtail[1],w=15,h=15,bg=0x0000)
+        draw.fill(x=self.body[-1][0]+1,y=self.body[-1][1]+1,w=13,h=13,bg=0xffff)
+
+        # if(self.justate):
+        #     self.justate = False
+        #     if len(self.body) == 1 or len(self.body) == 2:
+        #         # Here is movement after eating for a snake of size one
+        #         print("case 1")
+        #         print(self.body[-1][0])
+        #         print(self.body[-1][1])
+        #         draw.fill(x=self.body[-1][0],y=self.body[-1][1],w=13,h=13,bg=0xffff)
+        #     else:
+        #         # Here is movement after eating for a snake of size more than one
+        #         print("case 2")
+        #         draw.fill(x=self.body[-1][0],y=self.body[-1][1],w=13,h=13,bg=0xffff)
+        # else:
+        #     if len(self.body) == 1:
+        #         # Here is normal movement for a snake of size one
+        #         print("case 3")
+        #         print("erease old tail:")
+        #         print(self.body[0][0]-self.xdir)
+        #         print(self.body[0][1]-self.ydir)
+        #         print("draw new head")
+        #         print(self.body[0][0]+1)
+        #         print(self.body[0][1]+1)
+        #         draw.fill(x=(self.body[0][0]-self.xdir),y=(self.body[0][1]-self.ydir),w=15,h=15,bg=0x0000)
+        #         draw.fill(x=self.body[0][0]+1,y=self.body[0][1]+1,w=13,h=13,bg=0xffff)
+        #     else:
+        #         # Here is normal movement for a snake of size more than one
+        #         print("case 4")
+        #         print("erease old tail:")
+        #         print(self.body[0][0])
+        #         print(self.body[0][1])
+        #         print("draw new head")
+        #         print(self.body[-1][0])
+        #         print(self.body[-1][1])
+        #         # draw.fill(x=self.body[0][0],y=self.body[0][1],w=15,h=15,bg=0x0000)
+        #         draw.fill(x=self.oldtail[0],y=self.oldtail[1],w=15,h=15,bg=0x0000)
+        #         draw.fill(x=self.body[-1][0],y=self.body[-1][1],w=13,h=13,bg=0xffff)
```

Comment:
Blocks of commented out code should not be included in PRs. If you don't want it then delete it.

BTW `git gui` can be a good tool to selecting just the parts of the file you want to include in a commit (select what you want, right click and add lines).

If you